lntest/harness_assertion.go

```diff
@@ -1639,6 +1639,19 @@ func (h *HarnessTest) AssertPaymentStatus(hn *node.HarnessNode,
 	return target
 }
 
+// AssertPaymentFailureReason asserts that the given node lists a payment with
+// the given preimage which has the expected failure reason.
+func (h *HarnessTest) AssertPaymentFailureReason(hn *node.HarnessNode,
+	preimage lntypes.Preimage, reason lnrpc.PaymentFailureReason) {
+
+	payHash := preimage.Hash()
+	err := wait.Predicate(func() bool {
```

Comment:
future tips: we almost never use `Predicate` and favor `NoError` as it gives us more descriptive error - maybe we should remove it someday.